x-pack/plugin/esql/qa/testFixtures/src/main/java/org/elasticsearch/xpack/esql/EsqlTestUtils.java

```diff
@@ -185,7 +185,8 @@ public static List<List<Object>> getValuesList(Iterator<Iterator<Object>> values
 
     public static List<String> withDefaultLimitWarning(List<String> warnings) {
         List<String> result = warnings == null ? new ArrayList<>() : new ArrayList<>(warnings);
-        result.add("No limit defined, adding default limit of [500]");
+        result.add("No limit defined, adding default limit of [1000]");
+        result.add("No limit defined, adding default limit of [500]"); // this is for bwc tests, the limit in v 8.12.x is 500
```

Comment:
Same here about specific versions check.

Reply:
This specific check wasn't needed, it's never used by bwc tests